factory/gen.go

```diff
@@ -191,6 +193,20 @@ func (t *TLS) generateCert(secret *v1.Secret, cn ...string) (*v1.Secret, bool, e
 	return secret, true, nil
 }
 
+func IsExpired(secret *v1.Secret) bool {
+	certsPem := secret.Data[v1.TLSCertKey]
+	if len(certsPem) == 0 {
+		return false
+	}
+
+	certificates, err := cert.ParseCertsPEM(certsPem)
+	if err != nil || len(certificates) == 0 {
+		return false
+	}
+
+	return time.Now().After(certificates[0].NotAfter)
```

Comment:
Maybe the 90-day renewal point also needs to be considered?
`time.Now().Add(-time.Hour * 24 * ExpirationDaysCheck).After(certificates[0].NotAfter)`